pkg/client/core/api/rest/domain.go

```diff
@@ -12,6 +14,10 @@ type domainAPI struct {
 	client *HTTPClient
 }
 
+func (a *domainAPI) DeleteHostedZoneRecords(provider v1alpha1.CloudProvider, hostedZoneID string) error {
```

Comment:
This is very scary! Very, very, very scary. We need to be extremely careful, because we allow our users to reuse a hosted zone. There might be records in the zone that we do not "control". I don't think we want to do this with resources that we do not own. We should introduce an `okctl` tag and delete records related to that tag in which case.

Reply:
Hmmm --- maybe we should ask the user to manually delete any non SOA / NS records before running delete cluster ...